tests/test-endRequest.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+var http = require('http')
+  , request = require('../index')
+  , tape = require('tape')
+  , server
+
+
+tape('setup', function (t) {
+  server = http.createServer()
+  server.on('request', function (req, res) {
+    req.pipe(res)
+  })
+  server.listen(0, function() {
+    server.url = 'http://localhost:' + this.address().port
+    t.end()
+  })
+})
+
+tape('endRequest option', function (t) {
+  var testData = 'test request data'
+    , stream
+   
+  stream = request.post({
+    url: server.url,
+    endRequest: false
+  }, function(err, res, body) {
+    t.equal(res.statusCode, 200)
+    t.equal(body, testData + testData)
+    t.end()
+  })
+
+  setTimeout(function() {
+    stream.write(testData)
+  }, 10)
+
+  setTimeout(function() {
```

Comment:
Can't write after ending because the sole purpose of this test is to test manual ending after a few ticks. SetTimeouts replaced with setImmediate.